docs/roman/stpipe/user_pipeline.rst

```diff
@@ -0,0 +1,228 @@
+    suffix: null
+    - class: romancal.saturation.saturation_step.SaturationStep
+    ...
+
+
+Just like a ``Step``, it must have ``name`` and ``class`` values.
+Here the ``class`` must refer to a subclass of `stpipe.Pipeline`.
+
+Following ``name`` and ``class`` is the ``steps`` section.  Under
+this section is a subsection for each step in the pipeline.  The easiest
+way to get started on a parameter file is to call ``Step.export_config`` and
+then edit the file that is created.  This will generate an ASDF config file
+that includes every available parameter, which can then be trimmed to the
+parameters that require customization.
+
+For each Step’s section, the parameters for that step may either be
+specified inline, or specified by referencing an external
+parameter file just for that step.  For example, a pipeline
+parameter file that contains:
+
+.. code-block:: yaml
+
+    - class: romancal.jump.jump_step.JumpStep
+      name: jump
+      parameters:
+        flag_4_neighbors: true
+        four_group_rejection_threshold: 190.0
+        input_dir: ''
+        max_jump_to_flag_neighbors: 1000.0
+        maximum_cores: none
+        min_jump_to_flag_neighbors: 10.0
+
+is equivalent to:
+
+.. code-block:: yaml
+
+   steps:
+   - class: romancal.jump.jump_step.JumpStep
+     name: jump
+     parameters:
+        config_file = myjump.asdf
+
+with the file ``myjump.asdf.`` in the same directory:
+
+.. code-block:: yaml
+
+   class: romancal.jump.jump_step.JumpStep
+   name: jump
+   parameters:
+     flag_4_neighbors: true
+     four_group_rejection_threshold: 190.0
+
+If both a ``config_file`` and additional parameters are specified, the
+``config_file`` is loaded, and then the local parameters override
+them.
+
+Any optional parameters for each Step may be omitted, in which case
+defaults will be used.
+
+
+From Python
+-----------
+
+A pipeline may be configured from Python by passing a nested
+dictionary of parameters to the Pipeline’s constructor.  Each key is
+the name of a step, and the value is another dictionary containing
+parameters for that step.  For example, the following is the
+equivalent of the parameter file above:
+
+.. code-block:: python
+
+    from stpipe.pipeline import Image2Pipeline
+
+    steps = {
+        'jump':{'rejection_threshold': 180.,
+                'three_group_rejection_threshold': 190.,
+                'four_group_rejection_threshold':195.
+    }
+
+    pipe = ExposurePipeline(steps=steps)
+
+Running a Pipeline
+==================
+
+From the commandline
+--------------------
+
+The same ``strun`` script used to run Steps from the commandline can
+also run Pipelines.
+
+The only wrinkle is that any parameters overridden from the
+commandline use dot notation to specify the parameter name.  For
+example, to override the ``rejection_threshold`` value on the ``jump``
+step in the example above, one can do::
+
+    > strun romancal.pipeline.ExposurePipeline --steps.jump.rejection_threshold'=180.
```

Comment:
Stray apostrophe left in after threshold?

Reply:
Correct.